src/GENIEManager.cpp

```diff
@@ -0,0 +1,193 @@
+///////////////////////////////////////////////////////////////////////////////
+//
+// GENIEManager.cpp
+//
+// Manages GENIE input to qpixg4
+//
+// Created by: Ilker 20122021
+// Modified by: Dave Elofson 01092022
+//
+///////////////////////////////////////////////////////////////////////////////
+
+
+// Class Includes
+#include "GENIEManager.h"
+
+// Package Includes
+#include "ConfigManager.h"
+
+// ROOT Includes
+#include "TBranch.h"
+#include "TFile.h"
+#include "TTree.h"
+
+// Geant4 includes
+#include "globals.hh"
+#include "G4AutoLock.hh"
+
+// System Includes
+#include <iostream>
+#include <string>
+
+
+
+G4ThreadLocal GENIEManager * GENIEManager::instance_=0;
+
+//--------------------------------
+GENIEManager::GENIEManager()
+  : event_(0), NParticles_(0), f_(0), tree_(0), xsec_(0.),
+  dxsec_(0.), weight_(0.), prob_(0.), 
+  NParticles(0),
+  inputFile_(""), particleType_(""), genieFormat_(""), treeName_(""),
+  isInitialized(0)
+{
+    memset(pdg_, 0, Size_*sizeof(G4int));
+    memset(vtx_, 0, Size_*sizeof(G4double));
+    memset(X4_, 0, 4*Size_*sizeof(G4double));
+    memset(P4_, 0, 4*Size_*sizeof(G4double));
+    memset(polz_, 0, 3*Size_*sizeof(G4double));
+    memset(idx, 0, Size_*sizeof(G4int));
+    memset(Status, 0, Size_*sizeof(G4int));
+    memset(rescat_, 0, Size_*sizeof(G4int));
+    memset(FirstMother, 0, Size_*sizeof(G4int));
+    memset(LastMother, 0, Size_*sizeof(G4int));
+    memset(FirstDaughter, 0, Size_*sizeof(G4int));
+    memset(LastDaughter, 0, Size_*sizeof(G4int));
```

Comment:
In place of `memset()`, you can (if you want to!), use `std::fill()` using the array name; for example
```std::fill(pdg_, pdg+Size_, 0);```
The operator+() on a pointer will automatically step according to the size of the data type.

Reply:
keeping this for now